monero-wallet-gui.pro

```diff
@@ -144,26 +144,36 @@ CONFIG(WITH_SCANNER) {
 
 win32 {
 
+    # QMAKE_HOST.arch is unreliable, will allways report 32bit if mingw32 shell is run.
+    # Obtaining arch through uname should be reliable. This also fixes building the project in Qt creator without changes.
+    MSYS_HOST_ARCH = $$system(uname -a | grep -o "x86_64")
+
+    # Even if host is 64bit, we are building win32 spec, so mingw path should allways be "/mingw32"
```

Comment:
Because https://github.com/monero-project/monero-core/blob/master/monero-wallet-gui.pro#L145 species a win32 build.

The code you are quoting checks if the host in win32 or win64, as to make sure the msys path is correct (either c:/msys64 or c:/msys32)

Previously `contains(QMAKE_HOST.arch, x86_64) {` would always detect the host machines as win32, even when it is not.